src/reconstruction/refinement/removeFieldEntriesForType.m

```diff
@@ -114,6 +114,8 @@
                 end
             end              
         end
+        % Fix rules that now have more than one continuous "|"
+        model.rules = regexprep(model.rules, '\|{2,}', '|');
```

Comment:
Could you give me an example, where this is happening? 
Since I think it would be better to correct the initial replacements.

Reply:
This should recapitulate the problem (with Recon 2.04)
removeFieldEntriesForType(recon204, [35, 36], 'genes', numel(recon204.genes));

The following rules 47   361   362   363   364   365   366 have || or more.

Reply:
I see.
From a quick glance it looks as if the enclosing parenthesis cause this. Just replacing || could lead to odd situations. I'll have a close look at this.

Reply:
I have a (I think) more elegant solution using the Rules Parser.
The advantage of this would be that this code would become more independent of the actual format of the rules. 
Please have a look at my [GPRParserUpdate](https://github.com/tpfau/cobratoolbox/tree/GPRParserUpdate) branch. What do you think?